### PR TITLE
Building for arm64

### PR DESCRIPTION
**Pull request by agonzale34**
# Changelog
- Updating project to use Go Modules
- Updating building options to generate the files with Github actions and with binary for mac arm64

The testing release generated: https://github.com/Wattpad/boilr/releases/tag/0.3.0.1

When this gets merged to master I'll create Release for tag: `0.3.1` which will generate the binaries with the right version name and we will be able to point devs to install this version instead of the original one 

for local testing you should be able to do: `go mod tidy` to pull the dependencies and `go build` to generate the executable file, finally `./boilr version` you should see in the output something like:

```bash
 % ./boilr version
[i] Current version is 0.3.1
```

### CODE REVIEW

File: boilr.go [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 
 func main() {
-  if exists, err := osutil.DirExists(boilr.Configuration.TemplateDirPath); ! exists {
-    if err := osutil.CreateDirs(boilr.Configuration.TemplateDirPath); err != nil {
-      exit.Error(fmt.Errorf("Tried to initialise your template directory, but it has failed: %s", err))
-    }
-  } else if err != nil {
-    exit.Error(fmt.Errorf("Failed to init: %s", err))
-  }
+	if exists, err := osutil.DirExists(boilr.Configuration.TemplateDirPath); !exists {
```
Comment (Kieran-McC) [inline boilr.go:13]: Hmm whats going on here it looks like we didn't change anything of significance and we reformatted the file. If this isn't important I say we dont submit this change.

File: .idea/.gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (Kieran-McC) [inline .idea/.gitignore]: These are IDE auto generated files - we should remove this and the one below

File: .gitignore [hunk 3]
```diff
@@ -1,31 +1,121 @@
...
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 
-# don't accidentally commit a build of main
-boilr
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
```
Comment (Kieran-McC) [inline .gitignore]: I'm a little confused why we are adding all this to the gitignore?

Comment (agonzale34) [reply .gitignore]: Ohhh I'm using intellij idea as IDE and it always creates this folder, and I seem that other devs use goLand, so I wanted to add the recommended ignores for jetBreains idea. does wattpad has a standard ignore file for go projects?

File: boilr.go (hunk 1 above)
Comment (agonzale34) [reply boilr.go:13]: this was automatically reformated by Intellij idea, I checked with other files in this project and those are using `tabs` indentation, this file was using 2 spaces for it, so I think is good to keep these changes as they adjust that to use the project standard.

File: .gitignore (hunk 3 above)
Comment (Kieran-McC) [reply .gitignore]: I would just use ` .idea ` - you can look at the highlander gitignore for reference

Comment (agonzale34) [reply .gitignore]: sure, I'll do that.

File: .idea/.gitignore (hunk 2 above)
Comment (agonzale34) [reply .idea/.gitignore]: sure, I updated the changes to reflect this

**Review by Kieran-McC — APPROVED** [review]
Nice work @agonzale34

**Pull request merged**